### PR TITLE
API Integration for Challenges Page.

### PR DESCRIPTION
**Pull request by Ankita2002-Fr**
@whyDontI @sumitd94

I have added Accordion component for challenges. Please review the WIP PR.

### CODE REVIEW

**Comment by Ankita2002-Fr** [comment]
@whyDontI  @sumitd94 
I have worked on your comments and added all section code in pages/challenges itself. 

Please review the PR!

**Comment by Ankita2002-Fr** [comment]
@whyDontI @sumitd94 

I have modified the code in pages/challenges.
Please review the PR!

File: src/pages/challenges.tsx [hunk 1]
```diff
@@ -1,17 +1,77 @@
...
+import Active from '@/components/challenges/active';
+import Complete from '@/components/challenges/complete';
+import Accordion from '@/components/Accordion';
+import useFetch from '@/hooks/useFetch';
+import { challenge } from '@/components/constants/types';
+import classNames from '@/styles/tasks.module.scss';
 
+const CHALLENGES_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/challenges`;
+
+function renderCardList(challengeSection: challenge['content'], key:string) {
```
Comment (sumitd94) [inline src/pages/challenges.tsx:13]: arrow function?

File: src/pages/challenges.tsx [hunk 2]
```diff
@@ -1,17 +1,77 @@
...
+      challenges = response.challenges;
+      const challengeMap: any = [];
+      challenges.forEach((item) => {
+        if (item.is_active) {
+          if ('Active' in challengeMap) {
+            challengeMap.Active = [...challengeMap.Active, item];
+          } else {
+            challengeMap.Active = [item];
+          }
+        } else if (!item.is_active) {
```
Comment (sumitd94) [inline src/pages/challenges.tsx:38]: No need of `else if` here, if the condition in line no 32 fails, it should directly come here
so,
```javascript
else {...your logic}
```

File: src/pages/challenges.tsx [hunk 3]
```diff
@@ -1,17 +1,77 @@
...
+          if ('Completed' in challengeMap) {
+            challengeMap.Completed = [...challengeMap.Completed, item];
+          } else {
+            challengeMap.Completed = [item];
+          }
+        }
```
Comment (sumitd94) [inline src/pages/challenges.tsx:44]: I think the entire `if else` can be eliminated and you can make use of the `Array filter` method
Try checking this [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

**Review by sumitd94 — CHANGES_REQUESTED** [review]
Few changes requested please

**Comment by Ankita2002-Fr** [comment]
@whyDontI  @sumitd94  
I have worked on all comments and filtered the challenges.
Please review the PR!

**Review by whyDontI — APPROVED** [review]
LGTM 👍🏽 
Great work @Ankita2002-Fr 🥳

**Pull request merged**